cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCache.java

```diff
@@ -127,39 +144,73 @@ private void mergeItems(String type, Collection<CacheData> items) {
             }
         }
 
-        final String[] relationships = relationshipNames.toArray(new String[relationshipNames.size()]);
-        final String[] ids = idSet.toArray(new String[idSet.size()]);
-
+        int saddOperations = 0;
+        int msetOperations = 0;
+        int hmsetOperations = 0;
+        int pipelineOperations = 0;
+        int expireOperations = 0;
         if (keysToSet.size() > 0) {
             try (Jedis jedis = source.getJedis()) {
                 Pipeline pipeline = jedis.pipelined();
-                pipeline.sadd(allOfTypeReindex(type), ids);
-                pipeline.sadd(allOfTypeId(type), ids);
-
-                for (List<String> keys : Iterables.partition(keysToSet, maxMsetSize)) {
-                    pipeline.mset(keys.toArray(new String[keys.size()]));
+                for (List<String> idPart : Iterables.partition(idSet, options.getMaxSaddSize())) {
+                    final String[] ids = idPart.toArray(new String[idPart.size()]);
+                    pipeline.sadd(allOfTypeReindex(type), ids);
+                    saddOperations++;
+                    pipeline.sadd(allOfTypeId(type), ids);
+                    saddOperations++;
                 }
 
-                if (relationships.length > 0) {
-                    pipeline.sadd(allRelationshipsId(type), relationships);
+                for (List<String> keys : Lists.partition(keysToSet, options.getMaxMsetSize())) {
```

Comment:
Lists.partition vs Iterables.partition (was recommended in the Iterables javadoc)